.github/workflows/tests.yml

```diff
@@ -8,13 +11,59 @@ on:
   workflow_dispatch:
 
 jobs:
-  unittest-and-lint:
-    runs-on: ${{matrix.os}}
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+
+      - name: Setup up Linter
+        run: |
+          pip install toml yapf
+
+      - name: Lint code formatting
+        # make yapf fail if any file needs to be formatted
+        run: |
+          yapf -r --diff examples/ odxtools/ tests/
+
+  static-type-checking:
+    runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        # restrict the matrix to the oldest and the latest Python
+        # version being supported by odxtools
+        python-version: ["3.8", "3.11"]
```

Comment:
Maybe use `--python-version` argument of mypy instead?

https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-python-version

Reply:
I prefer the matrix notation and thus multiple job instances, because one can precisely tell, which runner (i.e. which version) failed. There is no need to have a look at the logs of Github Actions. It happens quite often, that certain typing annotations are supported with the latest version of Python but not with previous versions.